src/services/expense.service.js

```
@@ -0,0 +1,93 @@
+/* eslint-disable no-console */
+const { models } = require('../models/models');
+const Expense = models.Expense;
+const { Sequelize } = require('sequelize');
+// const { Expense } = require('../models/Expense.model.js');
+
+const getAll = async (userId, categories, from, to) => {
+  const data = {};
+
+  if (userId) {
+    data.userId = userId;
+  }
+
+  if (categories) {
+    data.category = categories;
+  }
+
+  if (from && to) {
+    data.spentAt = { [Sequelize.Op.between]: [from, to] };
+  }
+
+  try {
+    return Expense.findAll({ where: data });
+  } catch (error) {
+    console.log('Error getting expenses', error);
+    throw error;
+  }
+};
+
+const getById = async (id) => {
+  try {
+    return Expense.findByPk(id);
+  } catch (error) {
+    console.log(`Error getting expense with id ${id}`, error);
+    throw error;
+  }
+};
+
+const create = async (userId, spentAt, title, amount, category, note) => {
+  try {
+    return await Expense.create({
+      userId,
+      spentAt,
+      title,
+      amount,
+      category,
+      note,
+    });
+  } catch (error) {
+    console.log('Error creating expense', error);
+    throw error;
+  }
+};
+
+const update = async (id, title, amount, category, note) => {
+  const expense = getById(id);
+
```

You are checking if the expense exists, which is good, but because of the missing `await` in line 56 it may not work as expected. Consider handling the case where the expense doesn't exist after awaiting the `getById` call. For example, you might want to throw an error or return null.